packages/contracts/contracts/identity/IdentityProxy.s

```diff
@@ -0,0 +1,200 @@
+pragma solidity ^0.4.24;
+
+interface ERC725 {
+    event DataChanged(bytes32 indexed key, bytes32 indexed value);
+    event OwnerChanged(address indexed ownerAddress);
+    event ContractCreated(address indexed contractAddress);
+
+    function changeOwner(address _owner) external;
+    function getData(bytes32 _key) external view returns (bytes32 _value);
+    function setData(bytes32 _key, bytes32 _value) external;
+    function execute(uint256 _operationType, address _to, uint256 _value, bytes _data) external payable;
+}
+
+interface ERC20 {
+    function approve(address _spender, uint256 _value) public;
+}
+
+contract IdentityProxy is ERC725 {
+
+    uint256 constant OPERATION_CALL = 0;
+    uint256 constant OPERATION_CREATE = 1;
+
+    mapping(bytes32 => bytes32) store;
+
+    mapping(address => uint) public nonce;
+
+    // the owner
+    address public owner;
+
+    event Forwarded (bytes sig, address signer, address destination, uint value, bytes data, bytes32 _hash);
+
+    modifier onlyOwner() {
+        require(owner == 0x0 || msg.sender == owner, "only-owner-allowed");
+        _;
+    }
+
+    function () external payable {}
+
+    function changeOwner(address _owner)
+        external
+        onlyOwner
+    {
+        owner = _owner;
+        emit OwnerChanged(owner);
+    }
+
+    function getData(bytes32 _key)
```

Comment:
What are we planning to use getData and setData for ?

Reply:
These come as part of the ERC725 v2 standard. We could potentially use them for storing the IPFS profile instead of the EventLog method we're currently using. Since it's only one slot it shouldn't be too expensive.

Reply:
Sounds good.